analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -87,70 +86,55 @@ internal static class DiagnosticVerifier
+            GetAnalyzerDiagnostics(compilation, new[] { diagnosticAnalyzer }, CompilationErrorBehavior.FailTest);
+
+        public static ImmutableArray<Diagnostic> GetAnalyzerDiagnostics(
+            Compilation compilation,
+            IEnumerable<DiagnosticAnalyzer> diagnosticAnalyzers,
+            CompilationErrorBehavior checkMode,
             string sonarProjectConfigPath = null)
         {
             var supportedDiagnostics = diagnosticAnalyzers
                     .SelectMany(analyzer => analyzer.SupportedDiagnostics)
                     .Select(diagnostic => new KeyValuePair<string, ReportDiagnostic>(diagnostic.Id, ReportDiagnostic.Warn))
                     .Concat(new[] { new KeyValuePair<string, ReportDiagnostic>(AnalyzerFailedDiagnosticId, ReportDiagnostic.Error) });
+            var ids = supportedDiagnostics.Select(x => x.Key).ToHashSet();
 
             var compilationOptions = compilation.Options.WithSpecificDiagnosticOptions(supportedDiagnostics);
-            var actualToken = cancellationToken ?? CancellationToken.None;
             var analyzerOptions = string.IsNullOrWhiteSpace(sonarProjectConfigPath) ? null : TestHelper.CreateOptions(sonarProjectConfigPath);
-
             var diagnostics = compilation
                 .WithOptions(compilationOptions)
                 .WithAnalyzers(diagnosticAnalyzers.ToImmutableArray(), analyzerOptions)
-                .GetAllDiagnosticsAsync(actualToken)
+                .GetAllDiagnosticsAsync(default)
                 .Result;
 
-            if (!actualToken.IsCancellationRequested)
+            if (checkMode == CompilationErrorBehavior.FailTest)
             {
-                if (verifyNoException)
-                {
-                    VerifyNoExceptionThrown(diagnostics);
-                }
-                if (checkMode == CompilationErrorBehavior.FailTest)
-                {
-                    VerifyBuildErrors(diagnostics, compilation);
-                }
+                VerifyBuildErrors(diagnostics, compilation);
             }
-
-            return diagnostics;
-        }
-
-        internal static IEnumerable<Diagnostic> GetDiagnostics(Compilation compilation,
-            DiagnosticAnalyzer[] diagnosticAnalyzers, CompilationErrorBehavior checkMode,
-            bool verifyNoExceptionIsThrown = true,
-            string sonarProjectConfigPath = null)
-        {
-            var ids = diagnosticAnalyzers
-                .SelectMany(analyzer => analyzer.SupportedDiagnostics)
-                .Select(diagnostic => diagnostic.Id)
-                .Distinct()
-                .ToHashSet();
-
-            return GetAllDiagnostics(compilation, diagnosticAnalyzers, checkMode, verifyNoExceptionIsThrown, sonarProjectConfigPath: sonarProjectConfigPath)
-                .Where(d => ids.Contains(d.Id));
+            return diagnostics.Where(x => ids.Contains(x.Id)).ToImmutableArray();
```

Comment:
_Converted to draft:_ I need to validate if this is correct in one more scenario.

Reply:
It's good. No changes needed.